### PR TITLE
WASM, C, CPP: Subroutine intent unspecified

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
towards https://github.com/lfortran/lfortran/issues/1027

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
This fixes the issue in `wasm` backend shared at https://github.com/lfortran/lfortran/issues/1027#issue-1457182535.

File: src/libasr/codegen/asr_to_wasm.cpp [hunk 1]
```diff
@@ -1645,13 +1648,24 @@ class ASRToWASMVisitor : public ASR::BaseVisitor<ASRToWASMVisitor> {
...
+                wasm::emit_set_local(
+                    m_code_section, m_al,
+                    m_var_name_idx_map[get_hash((ASR::asr_t *)return_var)]);
+            } else if (ASR::is_a<ASR::ArrayItem_t>(*return_expr)) {
+                // emit_memory_store(ASRUtils::EXPR(return_var));
+
+                throw CodeGenError(
+                    "Passing array elements as arguments (with intent out, "
+                    "inout, unspecified) to Subroutines is not yet supported");
+            } else {
```
Comment (Shaikh-Ubaid) [inline src/libasr/codegen/asr_to_wasm.cpp:1666]: Passing array elements as (intent `out`/`inout`/`unspecified`) arguments to `subroutine` (as shared at https://github.com/lfortran/lfortran/issues/1027#issuecomment-1321403554) is not (yet) supported in the `wasm` backend. 

I attempted to support it in this PR, but (it seems) it is not being straightforward to support it at the moment. Temporarily the `wasm` backend throws a `CodeGenError()` as in the above code (Lines `1663` - `1665`).

**Comment by Shaikh-Ubaid** [comment]
This is ready. Please possibly review and please share feedback.

**Review by certik — APPROVED** [review]
This PR looks good to me and can be merged as is. Thanks!

**Comment by Shaikh-Ubaid** [comment]
Thank you for the approval.

**Pull request merged**